src/cast_core/src/RandomSampler.cpp

```diff
@@ -73,12 +78,14 @@ void RandomSampler::run() {
             auto statTracker = _activeCollectionScannerInstances > 0
                 ? config->readWithScanOperation.start()
                 : config->readOperation.start();
-            int index = config->collections.size() > 1 ? config->integerDistribution(_random) : 0;
-            auto cursor = config->collections[index].aggregate(config->pipeline,
-                                                               mongocxx::options::aggregate{});
-            for (auto doc : cursor) {
+            int collId = config->collections.size() > 1 ? config->collectionDistribution(_random) : 0;
+            int recordId = config->documentDistribution(_random);
+            auto maybe_result = config->collections[collId].find_one(bsoncxx::builder::stream::document{} << "_id" << recordId << bsoncxx::builder::stream::finalize);
+            if (maybe_result) {
                 statTracker.addDocuments(1);
-                statTracker.addBytes(doc.length());
+                statTracker.addBytes(maybe_result->view().length());
+            } else {
+                throw;
```

Comment:
can we throw a more explicit exception perhaps `std::domain_error` with a `what` indicating the id of the document that was expected to exist but didnt'?

Reply:
Fixed in 33cd8e4